src/plugins/discover/public/application/apps/main/components/doc_table/doc_table_wrapper.tsx

```diff
@@ -0,0 +1,240 @@
+   * Current shared item title
+   */
+  sharedItemTitle?: string;
+  /**
+   * Current data test subject
+   */
+  dataTestSubj: string;
+  /**
+   * Loading state
+   */
+  isLoading: boolean;
+  /**
+   * Filter callback
+   */
+  onFilter: DocViewFilterFn;
+  /**
+   * Sorting callback
+   */
+  onSort?: (sort: string[][]) => void;
+  /**
+   * Add columns callback
+   */
+  onAddColumn?: (column: string) => void;
+  /**
+   * Reordering column callback
+   */
+  onMoveColumn?: (columns: string, newIdx: number) => void;
+  /**
+   * Remove column callback
+   */
+  onRemoveColumn?: (column: string) => void;
+}
+
+export const DocTableWrapper = ({
+  render,
+  columns,
+  rows,
+  indexPattern,
+  onSort,
+  onAddColumn,
+  onMoveColumn,
+  onRemoveColumn,
+  sort,
+  onFilter,
+  useNewFieldsApi,
+  searchDescription,
+  sharedItemTitle,
+  dataTestSubj,
+  isLoading,
+}: DocTableProps) => {
+  const [minimumVisibleRows, setMinimumVisibleRows] = useState(50);
+  const [
+    defaultSortOrder,
+    hideTimeColumn,
+    isShortDots,
+    sampleSize,
+    filterManager,
+    addBasePath,
+  ] = useMemo(() => {
+    const services = getServices();
+    return [
+      services.uiSettings.get(SORT_DEFAULT_ORDER_SETTING, 'desc'),
+      services.uiSettings.get(DOC_HIDE_TIME_COLUMN_SETTING, false),
+      services.uiSettings.get(UI_SETTINGS.SHORT_DOTS_ENABLE),
+      services.uiSettings.get(SAMPLE_SIZE_SETTING, 500),
+      services.filterManager,
+      services.addBasePath,
+    ];
+  }, []);
+
+  const onSkipBottomButtonClick = useCallback(async () => {
+    // delay scrolling to after the rows have been rendered
+    const bottomMarker = document.getElementById('discoverBottomMarker');
+    const wait = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+    // show all the rows
+    setMinimumVisibleRows(rows.length);
+
+    while (rows.length !== document.getElementsByClassName('kbnDocTable__row').length) {
+      await wait(50);
+    }
+    bottomMarker!.focus();
+    await wait(50);
+    bottomMarker!.blur();
+  }, [setMinimumVisibleRows, rows]);
+
+  const renderHeader = useCallback(
+    () => (
+      <TableHeader
+        columns={columns}
+        defaultSortOrder={defaultSortOrder}
+        hideTimeColumn={hideTimeColumn}
+        indexPattern={indexPattern}
+        isShortDots={isShortDots}
+        onChangeSortOrder={onSort}
+        onMoveColumn={onMoveColumn}
+        onRemoveColumn={onRemoveColumn}
+        sortOrder={sort as SortOrder[]}
+      />
+    ),
+    [
+      columns,
+      defaultSortOrder,
+      hideTimeColumn,
+      indexPattern,
+      isShortDots,
+      onMoveColumn,
+      onRemoveColumn,
+      onSort,
+      sort,
+    ]
+  );
+
+  const renderRows = useCallback(
+    (rowsToRender: DocTableRow[]) => {
+      return rowsToRender.map((current) => (
+        <TableRow
+          key={`${current._index}${current._type}${current._id}${current._score}${current._version}${current._routing}`}
```

Comment:
```suggestion
          key={`${current._index}${current._type ?? ''}${current._id}${current._score}${current._version}${current._routing}`}
```

Reply:
This will help us having slightly better `ids` when `_type` is removed in 8.0, in case we don't get to adjust this place.